### DISCUSSION TITLE
No matching function for call to 'min(uint32_t&, unsigned int)'

### DISCUSSION THREAD

**Discussion by mirzafahad** [stm32duino/Arduino_Core_STM32#1582] [Q&A] [answered]
Hi,

I was using an AVR Arduino with Adafruit's E-Ink Display library. Everything was fine. When I switched to STM32 Arduino Core (using Nucleo-64 L053R8) it threw an error saying:

```
Adafruit_ACeP.cpp:131:53: error: no matching function for call to 'min(uint32_t&, unsigned int)'
  131 |     uint32_t numbytes = min(remaining, sizeof(block));
```

Seems like `min()` is not defined. For now, I added this line in their library:

```
#undef min
inline int min(uint32_t a, unsigned int b) { return ((a) < (b) ? (a) : (b)); }
```

That seems to work for now. But what would be the right way to fix this issue?

-Fahad

**Comment by fpistm** [comment, marked as answer]
Hi @mirzafahad,
The core uses the STL C++.
min, max,... require to compare same type.

The Adafruit's E-Ink Display library needs some updates to avoid this kind of issue and harden the code.
`uint32_t numbytes = min(remaining, (uint32_t)sizeof(block));`
You should open an issue to this repo to fix all min/max call.

**Reply by mirzafahad** [reply]
Make sense. Thanks.
